content/source/docs/cloud/integrations/service-now/service-catalog.html.md

```diff
@@ -0,0 +1,165 @@
+
+1. In ServiceNow, open the "Service Catalog > Variable Sets" table by searching for
+   "variable sets" in the left-hand navigation.
+1. Click "New" to add a Variable Set.
+1. Select "Single-Row Variable Set".
+    - Title: User-visible title for the variable set.
+    - Internal name: The internal name for the variable set.
+    - Order: The order in which the variable set will be displayed.
+    - Type: Should be set to "Single Row"
+    - Application: Should be set to "Terraform"
+    - Display title: Whether the title is displayed to the end user.
+    - Layout: How the variables in the set will be displayed on the screen.
+    - Description: A long description of the variable set.
+1. Click "Submit" to create the variable set.
+1. Find and click on the title of the new variable set in the Variable Sets
+   table.
+1. At the bottom of the variable set details page, click "New" to add a new
+   variable.
+  - Type: Should be "Single Line Text" for most variables, or "Masked" for
+    variables containing sensitive values.
+  - Question: The user-visible question or label for the variable.
+  - Name: The internal name of the variable. This must be derived from the name of the
+      Terraform or environment variable. Consult the table below to determine the
+      proper prefix for each variable name.
+  - Tooltip: A tooltip to display for the variable.
+  - Example Text: Example text to show in the variable's input box.
+1. Under the "Default Value" tab, you can set a default value for the variable.
+1. Continue to add new variables corresponding to the Terraform and environment
+   variables the configuration requires.
+
+When the Terraform integration applies configuration, it will map ServiceNow
+variables to Terraform and environment variables using the following convention.
+ServiceNow variables that begin with "sensitive_" will be saved as sensitive
+variables within Terraform Cloud.
+
+ServiceNow Variable Name | Terraform Cloud Variable
+--|--
+`tf_var_VARIABLE_NAME` | Terraform Variable: `VARIABLE_NAME`
+`tf_env_ENV_NAME` | Environment Variable: `ENV_NAME`
+`sensitive_tf_var_VARIABLE_NAME` | Sensitive Terraform Variable (Write Only): `VARIABLE_NAME`
+`sensitive_tf_env_ENV_NAME` | Sensitive Environment Variable (Write Only): `ENV_NAME`
+
+## Provision Infrastructure
+
+Once a Service Catalog and VCS repositories have been configured, ServiceNow
+users will be able to request infrastructure to be provisioned by Terraform
+Cloud.
+
+These requests will be fulfilled by Terraform Cloud, which will:
+
+1. Create a new workspace using the VCS repository provided by ServiceNow.
+1. Configure variables for that workspace, also provided by ServiceNow.
+1. Plan and apply the change.
+1. Report the results, including any outputs from Terraform, to ServiceNow.
+
+Once this is complete, ServiceNow will reflect that the Request Item has been
+provisioned.
+
+-> **Note:** Currently, the integration defaults to creating workspaces with
```

Comment:
We recently got a comment that this note is confusingly worded. I wonder if we can clarify it a bit, though to me it reads fine; I may be biased since I know what it's trying to say already.

Reply:
Yeah, I took a shot at wording some better words:
```
-> **Note:** The integration creates workspaces with
[auto-apply](../../workspaces/settings.html#auto-apply-and-manual-apply)
enabled. Terraform Cloud will queue an apply for these workspaces whenever
changes are merged to the associated VCS repositories. This is known as the
[VCS-driven run workflow](../../run/ui.html).
```

Reply:
Hm, this is better but it doesn't get the core point that this will affect *all* workspaces created from that repo. In the scenario where VCS-repo-1 has 10 workspaces attached, all 10 will run on commit, and they'll all have auto-apply enabled. So this is sort of a footgun warning.

Reply:
I added this sentence to the end of the above: "It is important to keep in mind
that all of the ServiceNow workspaces connected to a given repository will be
updated whenever changes are merged to the associated branch in that repository."